tutorials/basic-topics/kadane.md

```diff
@@ -0,0 +1,245 @@
+---
+title: 'Kadane Algorithm'
+description: 'Kadane 1D algorithm finds the maximum sum subarray in an array of integers. Kadane 2D algorithm finds the maximum sum subarray in a 2D array of integers.'
+hide_table_of_contents: false
+keywords:
+  - leetcode
+  - tutorial
+  - Kadane
+  - algorithm
+---
+
+import TutorialAuthors from '@site/src/components/TutorialAuthors';
+
+<TutorialAuthors names="@ShivaRapolu01"/>
+
+## Overview
+
+Kadane’s algorithm is an iterative algorithm in which we search for a maximum sum contiguous subarray within 1D array.
+
+## Example 1: [53 - Maximum Subarray](https://leetcode.com/problems/maximum-subarray/)
+
+```
+Given an integer array nums, find the contiguous subarray (containing at least one number) which has the largest sum and return its sum.
+
+A subarray is a contiguous part of an array.
+Input: nums = [-2,1,-3,4,-1,2,1,-5,4]
+Output: 6
+Explanation: [4,-1,2,1] has the largest sum = 6.
+```
+
+> A subarray is a contiguous part of an array maintaining the order of elements.            
+>[1,2,3] is a subarray of [1,2,3,4,5] but [1,3,2] & [1,3,5] are not.
+
+Consider an array of positive integers only, the maximum sum subarray will be the entire array itself. If the array contains negative integers only, then the maximum sum subarray will be the maximum element of the array. 
+It gets tricker when we have both positive and negative numbers. 
+
+If the array consists of positive integers(need not be all positive integers). Then there definitely exists a subarray which has positive sum, because I can choose one positive element in worst case and it would be my subarray with positive sum. 
+
+The main idea of Kadane's algorithm is to neglect the negative sum subarrays and take maximum among the positive sum subarrays.
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@ShivaRapolu01"/>
+
+```cpp
+int maxSubArraySum(vector<int>& nums)
```

Comment:
include full solution. don't just paste the function here. besides, the class member should be `maxSubArray` instead of `maxSubArraySum`